commit 63fe4d87140e0040077164fd7ea71e56445922ef (refs/changes/81/3481/1)
Author: Mike Zeller <mike.zeller@joyent.com>
Date:   2018-02-26T09:44:46-08:00 (1 year, 7 months ago)
    
    TRITON-179 AddNic should check for existence of ipv4_ips properly

diff --git a/lib/nics.js b/lib/nics.js
index e740a12..0a0fbc3 100644
--- a/lib/nics.js
+++ b/lib/nics.js
@@ -167,7 +167,11 @@ function addNic(req, res, next) {
         // anyways. If a user passed in a specific ip we will use that.
         assert.uuid(networkArg.ipv4_uuid, 'ipv4_uuid');
         networkUuid = networkArg.ipv4_uuid;
-        networkIp = networkArg.ipv4_ips[0];
+
+        if (networkArg.ipv4_ips) {
+            networkIp = networkArg.ipv4_ips[0];
+        }
+
         break;
     default:
         return next(new InvalidArgumentError(
